src/containers/IdeaLab/Ben/fusionChartDemo.js

```diff
@@ -1,121 +1,502 @@
+                        sshowvalues: "0",
+                        palettecolors: "#5D62B5, #979AD0",
+                        yaxisname: "Number of attempts",
+                        showmean: "1",
+                        meanIconRadius: "10",
+                        showmedianvalues: "0",
+                        theme: "fusion",
+                        meaniconshape: "polygon",
+                        meaniconsides: "2",
+                        meaniconradius: "2",
+                        showalloutliers: "1",
+                        outliericonsides: "20",
+                        outliericonalpha: "40",
+                        outliericonshape: "triangle",
+                        outliericonradius: "4",
+                        plotspacepercent: "60",
+                        plottooltext:
+                          '<b><u>Number of attemtps taken to complete "$label"</u>:</b><br> <br> Max: $maxDataValue<br> Min: $minDataValue<br> <br> Mean: $mean<br> <br> Q3: $Q3<br> Median: $median<br> Q1: $Q1'
+                      },
+                      categories: [
+                        {
+                          category:
+                            filteredAnalytics &&
+                            activitiesData &&
+                            analyticsData &&
+                            Array.from(
+                              new Set(
+                                Object.keys(filteredAnalytics)
+                                  .map(item => ({
+                                    label:
+                                      activitiesData[
+                                        analyticsData[filteredAnalytics[item]]
+                                          .activityKey
+                                      ].name
+                                  }))
+                                  .map(s => s.label)
+                              )
+                            ).map(label => ({
+                              label: label
+                            }))
+                        }
+                      ],
+                      dataset: [
+                        {
+                          seriesname: "Interquartile Range",
+                          data: numAttemptsArray
+                        }
+                      ]
+                    }
+                  }}
+                />
+              )}
+            </Fragment>
+          ) : (
+            <Fragment>
+              <Typography variant="h5">
+                Fetching from /analytics/jupyterSolutions
+              </Typography>
+              <Typography variant="h6">
+                Data with pathKey = {jupyterAnalyticsPathKey}:
+              </Typography>
+              <Typography variant="h6">Loading...</Typography>
+            </Fragment>
+          )}
+        </Fragment>
       </Fragment>
     );
   }
 }
 
-// Function to seperate the outliers from the list of data
+var timeTakenArray = [];
+var loading = true;
```

Comment:
are you treating `loading` as global variable in this file? can you try incorporate this to the state?